torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -76,13 +76,23 @@ inline void throw_error_out_requires_grad(const char* name) {
       "but one of the arguments requires grad.");
 }
 
-inline void throw_error_for_complex_autograd(const Tensor& tensor, const char* name) {
-  if (tensor.requires_grad()) {
-    TORCH_CHECK(!tensor.is_complex(), name,
-                " does not support automatic differentiation for outputs with complex dtype.");
+inline bool is_complex_autograd(const Tensor& tensor) {
+  return tensor.is_complex() && tensor.requires_grad();
+}
+
+inline bool is_complex_autograd(const TensorList& tensorlist) {
+  for (auto tensor: tensorlist) {
+    if (is_complex_autograd(tensor)) {
+      return true;
+    }
   }
+  return false;
 }
 
+// TODO: Once we have C++17, this can be a template function rather than a macro.
+#define THROW_ERROR_FOR_COMPLEX_AUTOGRAD( t, name ) \
```

Comment:
I'm not sure to follow what the benefit of having a macro is here? Since you do more function calls anyways?